pkg/query/remote_engine.go

```diff
@@ -0,0 +1,225 @@
+// Copyright (c) The Thanos Authors.
+// Licensed under the Apache License 2.0.
+
+package query
+
+import (
+	"context"
+	"io"
+	"time"
+
+	"github.com/go-kit/log"
+	"github.com/go-kit/log/level"
+	"github.com/pkg/errors"
+	"github.com/prometheus/prometheus/model/labels"
+	"github.com/prometheus/prometheus/promql"
+	"github.com/prometheus/prometheus/promql/parser"
+	"github.com/prometheus/prometheus/util/stats"
+	"github.com/thanos-community/promql-engine/api"
+
+	"github.com/thanos-io/thanos/pkg/api/query/querypb"
+	"github.com/thanos-io/thanos/pkg/store/labelpb"
+)
+
+// Opts are the options for a PromQL query.
+type Opts struct {
+	AutoDownSample        bool
```

Comment:
We typically write it like this:
```suggestion
	AutoDownsample        bool
```

Reply:
@fpetkovski you forgot about this :smile: